homeassistant/components/media_player/sonos.py

```diff
@@ -874,6 +887,15 @@ def set_volume_level(self, volume):
         """Set volume level, range 0..1."""
         self._player.volume = str(int(volume * 100))
 
+    @soco_error
+    def set_shuffle(self, shuffle):
+        """Enable/Disable shuffle mode."""
+        self._shuffle = shuffle
```

Comment:
Instead of setting `self._shuffle` here, get the actual play mode from the device during `update`.

Reply:
self._shuffle actually needs to be set here to persist the shuffle attribute (line 775)

Reply:
But instead of persisting the shuffle mode based on what we last set it to, we should get the current mode from sonos.

Reply:
I get your point. This is not corrected